src/textual/widgets/text_input.py

```diff
@@ -0,0 +1,172 @@
+from __future__ import annotations
+
+from rich.console import RenderableType
+from rich.text import Text
+
+from textual import events
+from textual._types import MessageTarget
+from textual.geometry import Size
+from textual.keys import Keys
+from textual.message import Message
+from textual.reactive import Reactive
+from textual.widget import Widget
+
+
+class TextInputBase(Widget):
```

Comment:
Trying to think ahead a little here by extracting some common keybinding/text navigation stuff into a base class. Not sure if it will work out, but that's my thinking.